File: problems/1938-minimum-operations-to-make-the-array-increasing/solution.c


int minOperations(int* nums, int numsSize){
    int ops = 0;
    int len = numsSize;
    int arr[len];
    for (int i = 0; i < len; i++) {
        arr[i] = nums[i];
    }
    for (int i = 1, prev = arr[0]; i < len; i++) {
        if (arr[i] <= prev) {
            ops += 1 + prev - arr[i];
            prev++;
        } else {
            prev = arr[i];
        }
    }
    return ops;
}

File: problems/1938-minimum-operations-to-make-the-array-increasing/solution.java
class Solution {
    public int minOperations(int[] nums) {
        int ops = 0;
        for (int i = 1, prev = nums[0]; i < nums.length; i++) {
            if (nums[i] <= prev) {
                ops += 1 + prev - nums[i];
                prev++;
            } else {
                prev = nums[i];
            }
        }
        return ops;
    }
}
